Fix compile warnings in NWNX code

log build nwnx

Scanning dependencies of target Administration
[ 20%] Building CXX object Plugins/Administration/CMakeFiles/Administration.dir/Administration.cpp.o
/home/tauredor/.local/share/nwndev/unified/Plugins/Administration/Administration.cpp: In member function ‘NWNXLib::Services::Events::ArgumentStack Administration::Administration::GetServerName(NWNXLib::Services::Events::ArgumentStack&&)’:
/home/tauredor/.local/share/nwndev/unified/Plugins/Administration/Administration.cpp:268:77: warning: unused parameter ‘args’ [-Wunused-parameter]
 Events::ArgumentStack Administration::GetServerName(Events::ArgumentStack&& args)

---
Scanning dependencies of target Damage
[ 26%] Building CXX object Plugins/Damage/CMakeFiles/Damage.dir/Damage.cpp.o
/home/tauredor/.local/share/nwndev/unified/Plugins/Damage/Damage.cpp: In member function ‘ArgumentStack Damage::Damage::DealDamage(ArgumentStack&&)’:
/home/tauredor/.local/share/nwndev/unified/Plugins/Damage/Damage.cpp:247:26: warning: catching polymorphic type ‘class std::exception’ by value [-Wcatch-value=]
     catch(std::exception e)

--- a/Plugins/Administration/Administration.cpp
+++ b/Plugins/Administration/Administration.cpp
@@ -265,7 +265,7 @@ Events::ArgumentStack Administration::SetServerName(Events::ArgumentStack&& args
     return Events::Arguments();
 }
 
-Events::ArgumentStack Administration::GetServerName(Events::ArgumentStack&& args)
+Events::ArgumentStack Administration::GetServerName(Events::ArgumentStack&&)
 {
     CExoString serverName = Globals::AppManager()->m_pServerExoApp->GetNetLayer()->GetSessionName();
     return Events::Arguments(serverName.CStr());

--- a/Plugins/Damage/Damage.cpp
+++ b/Plugins/Damage/Damage.cpp
@@ -240,11 +240,11 @@ ArgumentStack Damage::DealDamage(ArgumentStack&& args)
     int damagePower = Services::Events::ExtractArgument<int32_t>(args);
 
     int range = 0;
-    try 
+    try
     {
         range = Services::Events::ExtractArgument<int>(args);
     }
-    catch(std::exception e) 
+    catch(const std::runtime_error& e)
     {
         LOG_WARNING("NWNX_Damage_DealDamage() called from NWScript without final parameter. Please download the latest versions of NWNX scripts.");
     }
@@ -278,7 +278,7 @@ ArgumentStack Damage::DealDamage(ArgumentStack&& args)
         pEffect->SetInteger(k, positive[k] ? vDamage[k] : -1);
     pEffect->SetInteger(17, true); // combat damage
     // ... and apply it
-    
+
     //Check if ranged (this sets bRangedAttack internally)
     pEffect->SetInteger(18, !!range);
 

--- a/Plugins/SQL/Targets/SQLite.cpp
+++ b/Plugins/SQL/Targets/SQLite.cpp
@@ -185,6 +185,7 @@ void SQLite::PrepareString(int32_t position, const std::string& value)
 
 void SQLite::PrepareBinary(int32_t position, const std::vector<uint8_t> &value)
 {
+    (void)position; (void)value;
     ASSERT_FAIL_MSG("Binary format not implemented for SQLite");
 }
 

--- a/Plugins/Util/Util.cpp
+++ b/Plugins/Util/Util.cpp
@@ -501,7 +501,7 @@ ArgumentStack Util::SetInstructionLimit(ArgumentStack&& args)
     return Services::Events::Arguments();
 }
 
-ArgumentStack Util::GetScriptReturnValue(ArgumentStack&& args)
+ArgumentStack Util::GetScriptReturnValue(ArgumentStack&&)
 {
     int32_t retVal = 0;
 

